docs/operate/invocation.mdx

```diff
@@ -9,286 +9,309 @@ import Admonition from '@theme/Admonition';
+<Admonition type="tip" title={"Make any service call idempotent by using Restate"}>
+    By using Restate and an idempotency key, you can make any service call idempotent, without any extra code or setup.
+    This is a very powerful feature to ensure that your system stays consistent and doesn't perform the same operation multiple times.
+</Admonition>
 
-For example, using HTTP and `curl`:
+<details className={"grey-details"}>
+    <summary> Tuning retention time</summary>
+
+    <CH.Section>
+    You can tune the [retention time](focus://3) on a service-level by using the [Admin API](focus://1[15:28]) ([docs](/references/admin-api#tag/service/operation/modify_service)):
+    ```shell
+    curl -X PATCH localhost:9070/services/MyService \
+        -H 'content-type: application/json' \
+        -d '{"idempotency_retention": "2days"}'
+    ```
+    The [retention time](focus://3) is in [humantime format](https://docs.rs/humantime/latest/humantime/).
+    </CH.Section>
+
+</details>
+
+### Private services
+<CH.Section>
+    When registering an endpoint, every service is by default reachable via HTTP requests.
+
+    You can configure a service as `private`, such that you can't invoke it over HTTP, through the [Admin API](focus://1[15:28]) ([docs](/references/admin-api#tag/service/operation/modify_service)):
+
+    ```shell
+    curl -X PATCH localhost:9070/services/MyService \
+        -H 'content-type: application/json' \
+        -d '{"public": false}'
+    ```
+
+    You can revert it back to public with [`{"public": true}`](focus://3).
+
+    <Admonition type="info">
+        Private services can still be invoked by other handlers via the SDK.
+    </Admonition>
+
+</CH.Section>
+
+## Kafka subscriptions
+
+Restate can trigger handlers via Kafka events.
+Have a look at the [Java](/develop/java/kafka) and [TypeScript](/develop/ts/kafka) docs to learn how to set up your handler to receive Kafka events.
+
+<CH.Section>
+    - **Create subscriptions** for [a handler](focus://4) on [a topic](focus://3) via:
+        ```bash
+        curl localhost:9070/subscriptions -H 'content-type: application/json' \
+            -d '{
+                "source": "kafka://my-cluster/my-topic",
+                "sink": "component://MyService/Handle",
+                "options": {"auto.offset.reset": "earliest"}
+            }'
+        ```
+    The [options field](focus://5) is optional and accepts any configuration parameter from [librdkafka configuration](https://github.com/confluentinc/librdkafka/blob/master/CONFIGURATION.md).
+</CH.Section>
+- **List** the current subscriptions via:
+  ```bash
+  curl localhost:9070/subscriptions
+    ```
+    The creation and listing of subscriptions returns an identifier.
+<CH.Section>
+    - **Delete** a subscription with [its identifier](focus://1[45:78]) via:
+        ```bash
+        curl -X DELETE localhost:9070/subscriptions/018cd978aa9f77b5a80454bc060c2a77
```

Comment:
This does not look like the correct format of a subscription id. It should start with `sub_...`.